okhttp/src/main/java/okhttp3/internal/connection/RouteSelector.java

```diff
@@ -168,7 +172,11 @@ private void resetNextInetSocketAddress(Proxy proxy) throws IOException {
       inetSocketAddresses.add(InetSocketAddress.createUnresolved(socketHost, socketPort));
     } else {
       // Try each address for best behavior in mixed IPv4/IPv6 environments.
+      callEventListener.eventListener().dnsStart(callEventListener.call(), socketHost);
       List<InetAddress> addresses = address.dns().lookup(socketHost);
```

Comment:
Based on your own comments it looks like you just haven't bothered with the failure case yet.  But flagging that the empty check below should be a failure also.